Extended xenobiology mutation toxins

St0rmC4st3r opened the pull request:
## About The Pull Request
Added chemical reactions for most mutation toxins to be obtainable through green slime extracts.

## Why It's Good For The Game
Adds more fun to xenobio green slimes. People have fun lynching xenobio for making all the crew felinides, the entire crew having fun being IPCs and gathering in the AI core to pray to Omnissiah, more fun from sparking a racial holy war between humans and pissed off humanized non-human crew.
(No, the android mutation toxin is still only admin-spawnable.)

## New Recipes 
Unstable mutation toxin = Inject a green slime core with Radium.
Stable mutation toxin = Unstable mutation toxin + 10 Blood
Lizard = Unstable mutation toxin + 10 Liquid Gibs
Felinid = Unstable mutation toxin + 10 Fentanyl + 10 Impedrezene
Fly = Unstable mutation toxin + 10 Unstable Mutagen
Moth = Unstable mutation toxin + 10 Lipolicide
Podperson = Unstable mutation toxin + 10 EZ Nutriment
Golem = Unstable mutation toxin + 20 Liquid Adamantine
Abductor = Unstable mutation toxin + 10 Morphine + 10 Mute Toxin
IPC = Unstable mutation toxin + 20 Teslium
Skeleton = Advanced mutation toxin + 30 Milk + 30 Fluorosulfuric acid
Ashwalker = Advanced mutation toxin + 1 Lizard mutation toxin + 5 Entropic Polypnium
Shadow = Advanced mutation toxin + 30 Liquid Dark Matter + 10 Holy water
Plasmaman = Advanced mutation toxin + 60 Liquid Plasma + 20 Uranium

## Changelog
:cl:
add: Added reactions for most racial mutation toxins.
add: Added Liquid adamantine as well as respective solidification reaction.
tweak: Changed some of the suggested reactions to make a bit more sense than reactions suggested on the wiki.
/:cl:

St0rmC4st3r committed:
Made most of the mutation toxins obtainable through xenobio

Crossedfall commented:
Please fix the following error from travis:
```dm
FAIL: /datum/unit_test/reagent_id_typos 0s
	REASON #1: Unknown chemical id "/datum/chemical_reaction/facid" in recipe /datum/chemical_reaction/mutationtoxin/skeleton
```
https://travis-ci.com/BeeStation/BeeStation-Hornet/jobs/248529082#L1068-L1069

MarkSuckerberg commented on code/modules/reagents/chemistry/recipes/others.dm:
```diff
@@ -600,3 +600,92 @@
+	name = /datum/reagent/mutationtoxin/skeleton
+	id = /datum/reagent/mutationtoxin/skeleton
+	results = list(/datum/reagent/mutationtoxin/skeleton = 1)
+	required_reagents  = list(/datum/reagent/aslimetoxin = 1, /datum/reagent/consumable/milk = 1, /datum/chemical_reaction/facid = 1) //Because acid melts flesh off.
```
did you mean to write `/datum/reagent/toxin/acid/fluacid = 1` (fluorosulfuric acid) instead of `/datum/chemical_reaction/facid = 1`?

St0rmC4st3r replied:
Oh, right. Just was really sleepy when coding that. I'll fix it later today when I get home.

St0rmC4st3r committed:
Fixes skeleton toxin reaction and changes the ashwalker one

St0rmC4st3r commented:
Added recipes in the PR comment

zeskorion commented:
>benobio now gets a slightly neutered pride's mirror

Make these harder. I'd advise, against my best powergaming interests, you also remove zombi and golem. Golem can roll runic, which breaks the game.  Zombie? They break holoparasites

St0rmC4st3r commented:
So become inspired by my virology cures PR? say no more!

St0rmC4st3r commented:
Also, how exactly zombies break holoparas?

zeskorion commented:
Sentient zombies dont actually enter a death state unless gibbed or somesuch. Similar to memento mori.

Basically the holopara can just pull the body along and rampage. Best to leave that to rng with miners and their little known ruins

St0rmC4st3r commented:
Alright, no zombies, but I think golems can be fixed by requiring high quantities of liquid adamantine for it and excluding some 
golem subtypes from the rng.

St0rmC4st3r committed:
Disables zombie toxin

No zombies until holopara is fixed.

St0rmC4st3r committed:
Adds liquid adamantine for golem toxin

St0rmC4st3r committed:
Removes runic golems from random pool

Zesko says it breaks the game.

St0rmC4st3r committed:
Making stuff more expensive

St0rmC4st3r commented:
Should be fine now

MarkSuckerberg reviewed (approved):
Seems good codewise to me.

ike709 commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -432,6 +432,20 @@
 		return
+	if (race=="random")
+		to_chat(H, mutationtext)
+		H.set_species(pick(/datum/species/jelly/slime,
```
Instead of listing these twice in two separate procs, just add a single list var to the reagent

St0rmC4st3r replied:
fixed

St0rmC4st3r replied:
Okay, it is not fixed

ike709 commented:
Didn't have time to finish reviewing.

St0rmC4st3r committed:
Revert "Clears code of double listing"

This reverts commit b0362f90591eab566eec6d9ac23f27a48ff05264.

St0rmC4st3r committed:
Clears code of double listing

St0rmC4st3r committed:
Revert "Clears code of double listing"

This reverts commit b0362f90591eab566eec6d9ac23f27a48ff05264.

St0rmC4st3r commented:
Maybe also increase unstable toxin amount required to make specialized toxins so that it wouldn't be cheaper to just chug unstable.

St0rmC4st3r committed:
Merge remote-tracking branch 'St0rmC4st3r/mutation_toxins_ftw' into mutation_toxins_ftw

St0rmC4st3r committed:
FINALLY Makes unstable mutation toxin actually work

St0rmC4st3r committed:
Removes the double message when using unstable toxin

St0rmC4st3r committed:
Remove golems from unstable mutation pool

St0rmC4st3r commented:
Alright, I think it is good now. Tested, the reagents seem to react just fine. It seems balanced enough. You cannot just powergame golems from mutation toxin, soooo.... I think it is ready for merging after Travis finishes the build.

ike709 reviewed (changes requested):
Still bad code.

St0rmC4st3r committed:
Quality code

Makes the list of random races be actually in the reagent itself like Ike wanted it

St0rmC4st3r committed:
More bad code

Why the hell does this work????
